src/components/BoardList/BoardList.component.js

```diff
@@ -18,33 +23,81 @@ const BoardList = () => {
             };
             getBoardData();
         } catch (error) {
-            console.log(error.message);
+            console.log(error);
         }
     }, []);
 
     const buttonProperties = {
         text: "New Board",
     };
 
-    const addHandler = () => {
-        const newData = {
+    const autoFocus = () => document.querySelector("input").focus();
+
+    const handleCreate = () => {
+        setCreated(!created);
+    };
+
+    const handleSave = () => {
+        const boardName = text;
+        const date = String(new Date().toLocaleDateString());
+        let newData = {
             id: boards.length + 1,
-            name: `Project ${boards.length + 1}`,
-            date: "2021.07.31",
+            name: boardName,
+            date,
         };
+        const { name } = newData;
 
-        setBoards([...boards, newData]);
+        if (!name) {
+            setError("Board name is required");
+            autoFocus();
+        } else if (name.length > 30) {
+            setError("Only 30 characters allowed");
+            autoFocus();
+        } else {
+            setSuccess("Success!");
+            setTimeout(() => {
+                setBoards([...boards, newData]);
+                setCreated(!created);
+                setSuccess("");
+                setError("");
+                setText("");
+            }, 850);
+        }
+    };
+
+    const handleTyping = (e) => {
```

Comment:
Do not name a function that behavior refers to an other component. Tying not happening here, but in the form.
In this component you are handling the saving of data, its better to us handleBoradNameUpdate, Its more readable.
Think about moving this logic to the form itself (Your dashboard can be grow later, with a lot of functionality.)

Probably you should try to implememnt a context for boards...